snuba_sdk/legacy.py

```diff
@@ -66,14 +117,81 @@ def parse_exp(value: Any) -> Any:
     return Function(value[0], children, alias)
 
 
+def parse_automatic_condition(
+    col: str, values: Any, always_in: bool = False
+) -> Optional[Condition]:
+    """
+    Create an SDK condition using the values passed as extensions in the
+    legacy API.
+
+    :param col: The column that the automatic condition applies too.
+    :type col: str
+    :param values: The RHS values of the condition. Could be a single scalar
+        or a sequence of values.
+    :type values: Any
+    :param always_in: Some conditions always use an IN condition, even if there is a single value.
+    :type always_in: bool
+
+    """
+
+    column = Column(col)
+    if isinstance(values, int):
+        if always_in:
+            values = (values,)
+        else:
+            return Condition(column, Op.EQ, values)
+
+    if isinstance(values, list):
+        rhs: Sequence[Any] = tuple(map(parse_scalar, values))
+        return Condition(column, Op.IN, rhs)
+    elif isinstance(values, tuple):
+        rhs = tuple(map(parse_scalar, values))
+        return Condition(column, Op.IN, rhs)
```

Comment:
Am I missing something or both branches do the same thing ?

Reply:
No you're right. These branches originally had different logic and I must have forgotten to collapse them.